### PR TITLE
fix(typescript): Use consistent typescript configs - N8N-4552

### PR DESCRIPTION
**Pull request by netroy**
make all packages inherit tsconfig from the root tsconfig.

Ticket: N8N-4537

### CODE REVIEW

File: packages/cli/tsconfig.json [hunk 1]
```diff
@@ -1,27 +1,14 @@
...
-		"module": "commonjs",
-		"noImplicitAny": true,
-		"removeComments": true,
-		"strictNullChecks": true,
-		"forceConsistentCasingInFileNames": true,
-		"noImplicitReturns": true,
-		"preserveConstEnums": true,
-		"declaration": true,
-		"outDir": "./dist/",
-		"target": "es2017",
```
Comment (netroy) [inline packages/cli/tsconfig.json:20]: To support nodejs 14+, `"target": "es2020"` is recommended. But our current frontend tooling isn't handling `workflow` being compiled for `es2020`. Once we upgrade babel (or switch away from it), we can change the `target` across the code to `es2020`, as that'll produce smaller (and likely faster) JS files.

File: packages/cli/tsconfig.json [hunk 2]
```diff
@@ -1,27 +1,14 @@
...
-		"sourceMap": true,
-		"emitDecoratorMetadata": true,
-		"experimentalDecorators": true,
-		"esModuleInterop": true
+		// TODO: remove all options below this line
```
Comment (netroy) [inline packages/cli/tsconfig.json:9]: Leaving these `TODO` here because I'll be sending separate PRs removing these exceptions.

**Review by ivov — COMMENTED** [review]
Builds fine, runs fine, only minor comments.

1. Let's remove all tsx references? `"src/**/*.tsx"` `"tests/**/*.tsx",`, etc.

2. `useUnknownInCatchVariables` Comment to enable in future?

3. Settings like `strictNullChecks`, `noImplicitAny`, etc. are part of `strict`, which is enabled everywhere except cli. Maybe a comment to remove these settings that are part of `strict` once we have `strict` all over?

4. If we only need `emitDecoratorMetadata` and `experimentalDecorators` for TypeORM and for your router refactorings in cli, should we set them there only?

5. `"outDir": "dist"` is repeated in all packages. This is not inheritable?

Ideally if @alexgrozav can check the tsconfig for the FE packages.

**Comment by netroy** [comment]
> Builds fine, runs fine, only minor comments.
> 
>     1. Let's remove all tsx references? `"src/**/*.tsx"` `"tests/**/*.tsx",`, etc.

Done :heavy_check_mark: 

> 
>     2. `useUnknownInCatchVariables` Comment to enable in future?
enabled it globally, and disabled it on projects that needed it. So now the flag is under a `TODO` now.

>     3. Settings like `strictNullChecks`, `noImplicitAny`, etc. are part of `strict`, which is enabled everywhere except cli. Maybe a comment to remove these settings that are part of `strict` once we have `strict` all over?

I tried moving those to the `cli` tsconfig, but couldn't get it to work. I'll prioritize switching on `strict` for `cli`. we can remove these then. 

>     4. If we only need `emitDecoratorMetadata` and `experimentalDecorators` for TypeORM and for your router refactorings in cli, should we set them there only?


Done :heavy_check_mark: 

>     5. `"outDir": "dist"` is repeated in all packages. This is not inheritable?

[unfortunately not](https://github.com/microsoft/TypeScript/issues/29172). 
From what I understand, `outDir` is relative to the tsconfig defining it.

**Review by ivov — DISMISSED** [review]
Rebuilt, retested, all good. 

6. Do we need to build the tests? If I remove them from tsconfig, the test command runs fine, maybe ts-jest transpiles them on the fly.

7. When saving the tsconfigs locally, they are getting autoformatted with Prettier. If you can please format them and set up the extension `dangmai.workspace-default-settings` so your editor uses `.vscode/settings.default.json` to do this automatically in future.

Let's wait for Alex but otherwise LGTM. Thanks for cleaning this up.

**Comment by netroy** [comment]
> Rebuilt, retested, all good.
> 
>     6. Do we need to build the tests? If I remove them from tsconfig, the test command runs fine, maybe ts-jest transpiles them on the fly.

exactly, `ts-jest` transpiles the tests. we don't need to compile them.

>     7. When saving the tsconfigs locally, they are getting autoformatted with Prettier. If you can please format them and set up the extension `dangmai.workspace-default-settings` so your editor uses `.vscode/settings.default.json` to do this automatically in future.

I'll send another PR where I enable prettier prettier-eslint on everything. Will reformat all JSON files in there as well.

> 
> Let's wait for Alex but otherwise LGTM. Thanks for cleaning this up.

**Comment by ivov** [comment]
> we don't need to compile them.

Perfect, let's remove them all then from tsconfigs, to reduce build time and published package size.

**Comment by netroy** [comment]
> > we don't need to compile them.
> 
> Perfect, let's remove them all then from tsconfigs, to reduce build time and published package size.

done. also, ran prettier on all the tsconfigs.

**Review by ivov — DISMISSED** [review]
🚀

**Review by BHesseldieck — DISMISSED** [review]

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.194.0`

**Pull request merged**